File: 프로그래머스/1/12903. 가운데 글자 가져오기/가운데 글자 가져오기.py
def solution(s):
    if (len(s)%2)==0: #글자 길이가 짝수라면
        return s[len(s)//2-1:len(s)//2+1]
    
    else: #글자 길이가 홀수라면
        return s[len(s)//2]
        
    answer = ''
    return answer

File: 프로그래머스/1/12903. 가운데 글자 가져오기/가운데 글자 가져오기.java
class Solution{
    public String solution(String s){
        String answer="";
        int mid=s.length()/2;
        
        if(s.length()%2==1){
            answer=s.substring(mid,mid+1);
        }
        else{  //짝수
            answer=s.substring(mid-1,mid+1);
        }
        
        return answer;   
    }
}

/*
홀수
0 1 2 3 4 
mid=길이/2 =>2 , 2번인덱스 ok

짝수
0 1 2 3 4 5
mid=3번 + -1번인덱스
mid-1 ~ mid 
*/